Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py

```diff
@@ -1,27 +1,39 @@
 import FWCore.ParameterSet.Config as cms
 
+# HLT filter
 import HLTrigger.HLTfilters.hltHighLevel_cfi as hlt
-
 ALCARECOPPSCalMaxTracksFilter = hlt.hltHighLevel.clone(
   eventSetupPathsKey = 'PPSCalMaxTracks',
 )
 
-from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import *
+# EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff contains these ESSource modules:
+#   - totemDAQMappingESSourceXML_TrackingStrip
+#   - totemDAQMappingESSourceXML_TimingDiamond 
+#   - totemDAQMappingESSourceXML_TotemTiming 
+
+from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import totemDAQMappingESSourceXML_TimingDiamond 
+
+# in this minimal config we are not adding RECO to execution path
+# however if following import line is commented out (or removed) we have a crash and error message:
+#  `An entry in sequence pathALCARECOPromptCalibProdPPSDiamondSampicTimingCalib has no label`
+# does it mean that `PPSDiamondSampicTimingCalib` producer needs some objects from recoCTPPS_cff ?
+# before this PR AlCa Reco producer made by Laurent (PPSCalTrackBasedSel) was used, now it is disabled
+# this mean that CalibPPS/TimingCalibration/python/ALCARECOPPSCalTrackBasedSel_cff.py it not imported now
+# but ALCARECOPPSCalTrackBasedSel_cff.py has inside `from RecoPPS.Configuration.recoCTPPS_cff import *`
+# this may mean that `PPSDiamondSampicTimingCalib` has a hidden dependency on `PPSCalTrackBasedSel`
+# maybe all necessary imports should be added to `PPSDiamondSampicTimingCalib`
+from RecoPPS.Configuration.recoCTPPS_cff import *
 
-totemRPRawToDigi.rawDataTag = 'hltPPSCalibrationRaw'
-ctppsDiamondRawToDigi.rawDataTag = 'hltPPSCalibrationRaw'
-totemTimingRawToDigi.rawDataTag = 'hltPPSCalibrationRaw'
-ctppsPixelDigis.inputLabel = 'hltPPSCalibrationRaw'
+# raw2digi step
+from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsDiamondRawToDigi as _ctppsDiamondRawToDigi
+ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = cms.InputTag('hltPPSCalibrationRaw'))
 
-ctppsRawToDigiTask = cms.Task(
-  totemRPRawToDigi,
-  ctppsDiamondRawToDigi,
-  totemTimingRawToDigi,
-  ctppsPixelDigis
-)
+# rechits step
+from RecoPPS.Local.ctppsDiamondRecHits_cfi import ctppsDiamondRecHits as _ctppsDiamondRecHits
+ctppsDiamondRecHitsAlCaRecoProducer = _ctppsDiamondRecHits.clone(digiTag = 'ctppsDiamondRawToDigiAlCaRecoProducer')
```

Comment:
```suggestion
ctppsDiamondRecHitsAlCaRecoProducer = _ctppsDiamondRecHits.clone(digiTag = cms.InputTag("ctppsDiamondRawToDigiAlCaRecoProducer","TimingDiamond"))
```
Is it really necessary to change the name of the raw producer to ctppsDiamondRawToDigiAlCaRecoProducer instead of sticking with the defaults?

Reply:
Yes, the change of producer name to `ctppsDiamondRawToDigiAlCaRecoProducer ` is necessary. Unfortunately, it will propagate to all other modules,  see:
https://github.com/CTPPS/cmssw/pull/60/files